artifacts/registry.py

```diff
@@ -2,18 +2,62 @@
 """The artifact definitions registry."""
 
 from artifacts import definitions
+from artifacts import errors
+from artifacts import source_type
 
 
 class ArtifactDefinitionsRegistry(object):
   """Class that implements the artifact definitions registry."""
 
+  # yapf: disable
+  # yapf dedents the values of the key value pairs split across multiple lines.
+  _source_type_classes = {
+      definitions.TYPE_INDICATOR_ARTIFACT_GROUP:
+          source_type.ArtifactGroupSourceType,
+      definitions.TYPE_INDICATOR_COMMAND: source_type.CommandSourceType,
+      definitions.TYPE_INDICATOR_DIRECTORY: source_type.DirectorySourceType,
+      definitions.TYPE_INDICATOR_FILE: source_type.FileSourceType,
+      definitions.TYPE_INDICATOR_PATH: source_type.PathSourceType,
+      definitions.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY:
+          source_type.WindowsRegistryKeySourceType,
+      definitions.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE:
+          source_type.WindowsRegistryValueSourceType,
+      definitions.TYPE_INDICATOR_WMI_QUERY: source_type.WMIQuerySourceType,
+  }
+  # yapf: enable
+
   def __init__(self):
     """Initializes the artifact definitions registry object."""
     super(ArtifactDefinitionsRegistry, self).__init__()
     self._artifact_definitions = {}
     self._artifact_name_references = set()
     self._defined_artifact_names = set()
 
+  @classmethod
+  def CreateSourceType(cls, type_indicator, attributes):
```

Comment:
it seems odd that the registry is creating these objects. The registry should just return the correct class and the caller can create it. Call this "GetSourceTypeClass".